webknossos/webknossos/dataset/view.py

```diff
@@ -139,40 +281,130 @@ def read(
         ```
         """
 
-        offset = Vec3Int(offset)
-        size = self.size if size is None else Vec3Int(size)
+        current_mag_size: Optional[Vec3IntLike]
+        mag1_size: Optional[Vec3IntLike]
+        if absolute_bounding_box is None and relative_bounding_box is None:
+            if offset is None:
+                if size is None:
+                    assert (
+                        relative_offset is None and absolute_offset is None
+                    ), "You must supply size, when reading with an offset."
```

Comment:
the message makes sense, but at its execution `offset is None` is truthy, so something seems off to me?

Reply:
I'm not sure, what exactly seems off here? I guess this should be fine, since it fails in the case where `offset`, `size` and both bboxes are None, but one or both of `relative_offset` and `absolute_offset` are given. In this case, the size is missing, as we discussed in the chat.

Reply:
Ok, sorry, you are right of course. I think, I didn't negate the `assert` condition in my head and thought that the assertion is violated if `offset is None and size is None and relative_offset is None ...`, but it's obviously raised when `relative_offset is not None`. My bad!